maas/client/bones/__init__.py

```diff
@@ -296,6 +296,11 @@ def bind(self, **params):
         """
         data = dict(data)
         params = {name: data.pop(name) for name in self.handler.params}
+        # Extract the power parameters if they exist.
+        if 'power_parameters' in data:
+            power_parameters = data.pop('power_parameters')
+            for name, value in power_parameters.items():
+                data['power_parameters_%s' % name] = value
```

Comment:
I don't like how this is so specific to only power_parameters. The bones layer is not really supposed to know anything specific that is the viscera layer.

I would remove it this and below under line 206 in the diff, I would add.

```
for key, value in data.items():
    if isinstance(value, Mapping):
        del data[key]
        for nested_key, nested_value in value.items():
            data[key + '_' + nested_key] = nested_value
```